span.go

```diff
@@ -18,6 +18,10 @@ type spanImpl struct {
 	raw        RawSpan
 }
 
+var spanPool = &sync.Pool{New: func() interface{} {
```

Comment:
I kind of feel the same way, and changing `sync.Pool` to `*sync.Pool` in this one place would've fixed it all the same, but this having happened was reason enough to overcompensate a little.